ga4gh/backend.py

```diff
@@ -413,6 +429,18 @@ def getVariantSet(self, id_):
         dataset = self._getDatasetFromCompoundId(id_)
         return self.runGetRequest(dataset.getVariantSetIdMap(), id_)
 
+    def getVariant(self, id_):
+        """
+        Returns a variant with the given id
+        """
+        compoundId = variants.CompoundVariantId(id_)
+        dataset = self._getDatasetFromCompoundId(compoundId.datasetId)
+        variantSet = _safeMapQuery(
```

Comment:
Why not call ``dataset.getVariantSet(variantSetId)``, and push this functionality down into the dataset? It feels like we're moving a lot of maps around where we could be using an interface (and keeping the implementation flexible)?